hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/bulk/BulkInsertWriteFunction.java

```diff
@@ -113,7 +116,8 @@ public BulkInsertWriteFunction(Configuration config, RowType rowType) {
   public void open(Configuration parameters) throws IOException {
     this.taskID = getRuntimeContext().getIndexOfThisSubtask();
     this.writeClient = FlinkWriteClients.createWriteClient(this.config, getRuntimeContext());
-    this.ckpMetadata = CkpMetadata.getInstance(config);
+    HoodieTableMetaClient metaClient = StreamerUtil.createMetaClient(this.config);
+    this.ckpMetadata = CkpMetadataFactory.getCkpMetadata(metaClient, writeClient.getConfig(), config);
```

Comment:
I think it shoud be feasible if we only maintains the interfafce as `getCkpMetadata(HoodieWriteConfig writeConfig, Configuration conf)`, then there is no need to create the meta client.

Reply:
Yes, we can create the `CkpMetadata` only by `HoodieWriteConfig ` and `Configuration `. I've updated the code as you suggested.